consult-search.el

```diff
@@ -0,0 +1,92 @@
+;;; consult-search.el --- Providing search via Consult and Projectile  -*- lexical-binding: t; -*-
```

Comment:
I would like if it could have a less generic name. consult-project-grep or something like this, but no need to change it now - we might come up with something better.

Furthermore I think you can put it directly in consult.el and somehow get rid of the projectile dependency.